src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs

```diff
@@ -1,12 +1,202 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Tags;
+using Microsoft.VisualStudio.Core.Imaging;
+using Microsoft.VisualStudio.Imaging;
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
 {
     internal static class GlyphExtensions
     {
+        public static ImageId GetImageId(this Glyph glyph)
+        {
+            // VS for mac cannot refer to ImageMoniker
+            // so we need to expose ImageId instead of ImageMoniker here
+            // and expose ImageMoniker in the EditorFeatures.wpf.dll
+            switch (glyph)
+            {
+                case Glyph.None:
+                    return default;
+
+                case Glyph.Assembly:
+                    return new ImageId(KnownMonikers.Assembly.Guid, KnownMonikers.Assembly.Id);
```

Comment:
We have to be careful here: if this is embedding constants we're OK; if this is embedding references to the KnownMonikers type then that might cause mono to load ImageMoniker and thus still explode. Can we use KnownImageIds instead?